packages/docs-components/src/knobs/defaultComponents.tsx

```diff
@@ -49,6 +50,34 @@ const KnobSelect: React.FunctionComponent<KnobComponentProps> = props => (
   </select>
 )
 
+const KnobRange: React.FunctionComponent<KnobComponentProps> = props => {
+  const parseValue = (parseValue: string): number => {
+    const hasDecimal = /\.\d/.test(parseValue)
+
+    return hasDecimal ? parseFloat(parseValue) : parseInt(parseValue, 10)
+  }
+  const { defaultValue, unit } = React.useMemo(() => {
+    return {
+      defaultValue: parseValue(props.value),
+      unit: `${props.value}`.replace(`${parseValue(props.value)}`, ''),
+    }
+  }, [])
```

Comment:
Was taken from `KnobScalar` 🦅

Reply:
Cool, but looks like an unnecessary premature optimisation.

Reply:
Actually, no. It computes `defaultValue` & `unit` once because `defaultValue` depends on `props.value` which will be changed during renders